base/src/main/java/io/fabric8/launcher/base/JsonUtils.java

```diff
@@ -0,0 +1,84 @@
+package io.fabric8.launcher.base;
+
+import javax.json.JsonArrayBuilder;
+import javax.json.JsonObjectBuilder;
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.util.Map;
+
+import static javax.json.Json.createArrayBuilder;
+import static javax.json.Json.createObjectBuilder;
+
+public abstract class JsonUtils {
+
+    private JsonUtils() {
+    }
+
+    /**
+     * Converts a <code>Map</code> of type <code>&lt;String, Object&gt;</code>
+     * to a <code>JsonObject</code>. The values of the object can itself be
+     * objects, arrays or simple values
+     */
+    @SuppressWarnings("unchecked")
+    public static JsonObjectBuilder mapToJson(Map<String, Object> map) {
+        JsonObjectBuilder builder = createObjectBuilder();
+        for (Map.Entry<String, Object> entry : map.entrySet()) {
+            if (entry.getValue() instanceof Map) {
+                builder.add(entry.getKey(), mapToJson((Map<String, Object>) entry.getValue()));
+            } else if (entry.getValue() instanceof Iterable) {
+                builder.add(entry.getKey(), listToJson((Iterable<Object>)entry.getValue()));
+            } else if (entry.getValue() == null) {
+                builder.addNull(entry.getKey());
+            } else if (entry.getValue() instanceof Boolean) {
+                builder.add(entry.getKey(), (Boolean)entry.getValue());
+            } else if (entry.getValue() instanceof Double) {
+                builder.add(entry.getKey(), (Double)entry.getValue());
+            } else if (entry.getValue() instanceof Long) {
+                builder.add(entry.getKey(), (Long)entry.getValue());
+            } else if (entry.getValue() instanceof Integer) {
+                builder.add(entry.getKey(), (Integer)entry.getValue());
+            } else if (entry.getValue() instanceof BigInteger) {
+                builder.add(entry.getKey(), (BigInteger)entry.getValue());
+            } else if (entry.getValue() instanceof BigDecimal) {
+                builder.add(entry.getKey(), (BigDecimal) entry.getValue());
+            } else {
+                builder.add(entry.getKey(), entry.getValue().toString());
+            }
+        }
+        return builder;
+    }
+
+    /**
+     * Converts any <code>Iterable</code> of type <code>&lt;Object&gt;</code>
+     * to a <code>JsonArray</code>. The items of the array can itself be objects,
+     * arrays or simple values
+     */
+    @SuppressWarnings("unchecked")
+    public static JsonArrayBuilder listToJson(Iterable<Object> list) {
```

Comment:
Technically the parameter is not a `List`, but an `Iterable` :) 

Perhaps just name both methods `toJson`?

Reply:
Also, a test class would be nice

Reply:
One solution would be to pass the JsonArrayBuilder as a parameter here

Reply:
I had that before, but I didn't really like it. To me `toXxxx()` methods should be part of an actual object (like `path.toFile()` or `obj.toString()`), not as static methods. You lose a bit of context that way I think. I can change the `Iterable` to `List` though ;-)

Reply:
`Iterable` is fine, it's just the method name that may need to be changed.

Reply:
But I think passing the object as a parameter would be much better, what if you want to add that result to an existing builder?

Reply:
That seems like a strange thing to do. Do you have a use-case for that? And even so I'd say that it's easy enough to add overloads adding that extra parameter if at some future date we need that, right?

Reply:
I don't think it's strange, but I can live with that 👍